sympy/solvers/solveset.py

```diff
@@ -11,7 +11,9 @@
 
 from sympy.core.sympify import sympify
 from sympy.core import S, Pow, Dummy, pi, Expr, Wild, Mul, Equality
-from sympy.core.numbers import I, Number, Rational, oo
+from sympy.core.numbers import (I, Number, Rational, oo,
+                                               NegativeInfinity, Infinity,
```

Comment:
Since `S` is already imported at line 13 (and you use it to access the singletons) there is no need to import the constants here as `Infinity`, etc...)